### PR TITLE
Application custom config

### PR DESCRIPTION
**Pull request by jworkmanjc**
## Issues
* [CUT-3922](https://jumpcloud.atlassian.net/browse/CUT-3922) - Add additional application (sso) properties to 

## What does this solve?

The default models defined in our API specification are missing several properties in the Application model, This release manually re-defines the `config` model for the Application model.

## Is there anything particularly tricky?

I've added a new "find and replace" type of way to redefine the generated code. Instead of using regex, to find and replace many things, you just need to replace sections of the API model. Within the `apiTransform.ps1` file I've added a `OverrideDefinitions` list. Within that list, one can denote a part of the SDK swagger spec to replace entirely. All you would need to do that would be to define a `.json` file with the same name in the `/Custom` directory. 

## How should this be tested?

Using the current version of the V2 SDK (0.0.43) running the command: `remove-JCSystemGroupMember -SystemID 62ace7fbac30e538a6ba0ec8 -Groupid 604938c5232e1179d1e7bbdb` should return multiple status messages
Screenshot 2024-03-28 at 9 31 12 AM

Using this version of the V2 SDK (0.0.44) running the command: `remove-JCSystemGroupMember -SystemID 62ace7fbac30e538a6ba0ec8 -Groupid 604938c5232e1179d1e7bbdb` should return the single status message as expected
Screenshot 2024-03-28 at 9 30 45 AM

503 error should still catch and recover 😎
Screenshot 2024-03-28 at 9 32 43 AM

## Screenshots

[CUT-3922]: https://jumpcloud.atlassian.net/browse/CUT-3922?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ApiTransform.ps1 [hunk 1]
```diff
@@ -611,7 +617,7 @@ Function Update-SwaggerObject {
                         } Else {
                             $excludePath = ($InputObjectName -replace '.paths.', '') -replace "\.\w+", ''
                             # Write-Warning "[status] New SDK Endpoint Found: `nOperationId: $($ThisObject.operationId) `nPath: $excludePath"
-                            Write-Error ("In '$($CurrentSDKName)' unknown operationId '$($ThisObject.operationId) - $($InputObjectName)'.")
+                            Write-Warning ("In '$($CurrentSDKName)' unknown operationId '$($ThisObject.operationId) - $($InputObjectName)'.")
```
Comment (jworkmanjc) [inline ApiTransform.ps1:620]: This really didn't need to throw an error/ switched to write-warning

File: ApiTransform.ps1 [hunk 2]
```diff
@@ -69,6 +69,9 @@ $TransformConfig = [Ordered]@{
             '\[,'                                                                                                                                                                                                                   = '[';
             ',]'                                                                                                                                                                                                                    = ']';
         };
+        OverrideDefinitions = @(
```
Comment (jworkmanjc) [inline ApiTransform.ps1:72]: the idea here is that if you want to replace an entire section of the Swagger spec you could do that by specifying the path of item you wish to replace and adding a corresponding json file in `/CustomDefinitions/` directory

**Review by gweinjc — DISMISSED** [review]
Looks good to me, nice job with the added method of replacing things in the future if needed. I'm sure that'll come in handy!

**Review by kmaranionjc — DISMISSED** [review]

**Review by gweinjc — DISMISSED** [review]

**Review by gweinjc — APPROVED** [review]
Looks good, nice job with the custom definitions. It'll make future work much easier

**Review by kmaranionjc — APPROVED** [review]
Nice work! Looks good on my end

**Pull request merged**